### ISSUE TITLE
GUI stuck on "device is now in recovery mode"

### ISSUE THREAD

**Issue by daukstsoskars** [checkra1n/BugTracker#462]
**What device + iOS version are you on?**
iPad Air - iOS 12.4.3

**What checkra1n version are you using?**
v0.9.5

**What are the steps to reproduce the issue?**
1. Connect iPad to MacOS running Mojave
2. Trust device in iTunes
3. Run checkra1n
4. On welcome screen, click 'Start'
5. On the next screen, click 'Next'
6. Device boots in recovery mode
7. iTunes opens up with a window "There is a problem with the iPad <Name of ipad> that requires it to be updated or restored"
8. Go back to checkra1n GUI, see the message "Device is now in recovery mode". This message appears on the same page as the button we clicked in step 5.
9. GUI never goes to screen with manual steps of how to enter DFU mode.

**What do you expect, and what is happening instead?**
After device enters recovery mode GUI should go to the next screen with manual steps of how to enter DFU mode and then jailbreak the iPad.

**Any other info, error logs, screenshots, ...?**

Checkra1n gets stuck in this screen.
<img width="481" alt="Screen Shot 2019-11-18 at 2 16 07 PM" src="https://user-images.githubusercontent.com/37187829/69093416-651d4300-0a1c-11ea-8ba2-d462b4caf24a.png">

Itunes message after device enters recovery mode.
<img width="491" alt="Screen Shot 2019-11-18 at 2 16 00 PM" src="https://user-images.githubusercontent.com/37187829/69093417-651d4300-0a1c-11ea-874b-8e5313cddacf.png">

I have tried closing the itunes.
I have tried killing itunes process.
I have tried entering DFU mode manually even though GUI did not direct me to do so.

**Comment by tomasz-poplawski** [comment]
I have exactly the same problem on iPad Air and Mojave. Any solution?

**Comment by raymondwebb** [comment]
i have the same issue using ipad air on Catalina

**Comment by neikxd** [comment]
Yes, I have the same issue. iPad Air 1, iOS 12.4 + Mac 10.12 Sierra

**Comment by stevenb9** [comment]
Same issue, iPad Air - 12.4.3, Catalina 10.15.1. Tested 3 different macs

**Comment by johansci** [comment]
Run "./checkra1n.app/Contents/MacOS/checkra1n_gui -" in the terminal after you've entered the device into DFU mode. Fixed the issue for me.

**Comment by neikxd** [comment]
> Run "./checkra1n.app/Contents/MacOS/checkra1n_gui -" in the terminal after you've entered the device into DFU mode. Fixed the issue for me.

I guess the scheme like:
- enter device to DFU mode
- connect device to Mac
- run command in terminal

Is that all right?

**Comment by johansci** [comment]
> > Run "./checkra1n.app/Contents/MacOS/checkra1n_gui -" in the terminal after you've entered the device into DFU mode. Fixed the issue for me.
> 
> I guess the scheme like:
> 
> * enter device to DFU mode
> * connect device to Mac
> * run command in terminal
> 
> Is that all right?

I connected the device to Mac, executed command in terminal and then entered DFU mode.
I presume it continuously checks for device in DFU mode so should not matter which order.

**Comment by kappaHomo** [comment]
Hey, I had the same problem and i got solved by downloading 0.9.3 beta and run checkra1n before i got the ipad in DFU mode and connected to the pc.

**Comment by neikxd** [comment]
checkra1n 0.9.6 beta and lower don't support iPad Air (info from checkra.in). Just wait!

**Comment by johansci** [comment]
> checkra1n 0.9.6 beta and lower don't support iPad Air (info from checkra.in). Just wait!

I jailbroke my iPad Air with version 0.9.6

**Comment by neikxd** [comment]
I just confirm that: checkra1n 0.9.6 works for me.

On my iPad Air 1, iOS 12.4: 
- Run checkra1n in terminal: /checkra1n.app/Contents/MacOS/checkra1n
- Open GUI checkra1n app → "start" → "next" to go to Recovery mode
- Enter DFU mode by press "power button" + "home button" then release "power button" keep press "home button" for at least 8s
- checkra1n in terminal will do the job

Good luck!

**Comment by TommyySin** [comment]
> > checkra1n 0.9.6 beta and lower don't support iPad Air (info from checkra.in). Just wait!
> 
> I jailbroke my iPad Air with version 0.9.6

what version is your iPad Air?

**Comment by johansci** [comment]
> > > checkra1n 0.9.6 beta and lower don't support iPad Air (info from checkra.in). Just wait!
> > 
> > 
> > I jailbroke my iPad Air with version 0.9.6
> 
> what version is your iPad Air?

version 12.4.3

**Comment by PeterHdd** [comment]
I have tried the solutions above, and it is still not working.. downloaded 0.9.6 - 0.9.2 - 0.9.3 - 0.9.6 all didnt work. 

Im using ipad air with 12.4.3 ios, but nothing worked. Even when I right click and click on "Open package contents" and then on "Contents" -> "Mac OS" -> " checkra1n Terminal" still doesnt work.

How to solve this and make it work on 12.4.3??

**Comment by firdausalif** [comment]
> Run "./checkra1n.app/Contents/MacOS/checkra1n_gui -" in the terminal after you've entered the device into DFU mode. Fixed the issue for me.

this is fix my issue. Thanks bro

**Comment by amorrn** [comment]
I was able to use version 0.9.6 beta with my iPad Air 1 on 12.4.4 using the below sequence:

- Set device into DFU mode using the hardware keys
- Run /checkra1n.app/Contents/MacOS/checkra1n in the terminal

It didn't exit in the terminal after the exploit was completed but did show confirmation. I simply exited the program with ctrl-c after it rebooted into normal mode and the checkrain app showed up on my device.

**Comment by meigoulouseiy** [comment]
Or when iPad stuck in recovery mode and can't restore, force factory reset it, and check if it can be restored then. Or use EelPhone iPS System Repair tool to [fix iPad stuck in recovery mode](https://www.eelphone.com/solution/ipad-recovery-mode.html) without losing data.
